apis/Google.Cloud.AspNetCore.DataProtection.SecretManager/Google.Cloud.AspNetCore.DataProtection.SecretManager/CloudSecretManagerXmlRepository.cs

```diff
@@ -0,0 +1,128 @@
+
+    internal CloudSecretManagerXmlRepository(SecretManagerServiceClient client, string secretName, string projectName)
+    {
+        _secretManagerServiceClient = GaxPreconditions.CheckNotNull(client, nameof(client));
+        GaxPreconditions.CheckNotNull(secretName, nameof(secretName));
+        GaxPreconditions.CheckNotNull(projectName, nameof(projectName));
+        _projectName = new ProjectName(projectName);
+        _secretName = new SecretName(projectName, secretName);
+
+        // Version is populated in such a way that it always refers to the most latest version.
+        // For more information on version naming refer to
+        // https://cloud.google.com/dotnet/docs/reference/Google.Cloud.SecretManager.V1/latest/Google.Cloud.SecretManager.V1.AccessSecretVersionRequest#Google_Cloud_SecretManager_V1_AccessSecretVersionRequest_Name
+        _version = SecretVersionName.FromProjectSecretSecretVersion(projectName, secretName, "latest");
+    }
+
+    /// <inheritdoc />
+    public IReadOnlyCollection<XElement> GetAllElements()
+    {
+        if (!IsSecretExists())
+        {
+            return Array.Empty<XElement>();
+        }
+        var accessSecretVersionResponse = GetSecretVersionResponse();
+
+        if (accessSecretVersionResponse != null)
+        {
+            var secretPayload = accessSecretVersionResponse.Payload.Data.ToStringUtf8();
+            XDocument document = XDocument.Parse(secretPayload);
+            var res = document?.Root?.Elements().ToList();
+            return res ?? (IReadOnlyCollection<XElement>) Array.Empty<XElement>();
+        }
+        return Array.Empty<XElement>();
+
+        AccessSecretVersionResponse GetSecretVersionResponse()
+        {
+            var response = _secretManagerServiceClient.AccessSecretVersion(_version, _callSettings);
+            return response;
+        }
+    }
+
+    /// <inheritdoc />
+    public void StoreElement(XElement element, string friendlyName)
+    {
+        if (!IsSecretExists())
+        {
+            CreateSecret();
+        }
+        XDocument document = new XDocument(new XElement("root"));
+        document.Root.Add(element);
+        SecretPayload payload = new SecretPayload
+        {
+            Data = ByteString.CopyFrom(document.ToString(), Encoding.UTF8)
+        };
+        _secretManagerServiceClient.AddSecretVersion(_secretName, payload, _callSettings);
+    }
+
+    private void CreateSecret()
+    {
+        var createSecretRequest = new CreateSecretRequest
+        {
+            Secret = new Secret
+            {
+                Replication = new Replication
+                {
+                    Automatic = new Replication.Types.Automatic(),
+                },
+            },
+            ParentAsProjectName = _projectName,
+            SecretId = _secretName.SecretId
+        };
+        _secretManagerServiceClient.CreateSecret(createSecretRequest, _callSettings);
+    }
+
+    private bool IsSecretExists()
```

Comment:
The name sounds grammatically incorrect. Maybe we can use just `SecretExists`

Reply:
done